nomad/state/state_store.go

```diff
@@ -2212,11 +2212,13 @@ func (s *StateStore) CSIVolumesByNodeID(ws memdb.WatchSet, nodeID string) (memdb
 	iter := NewSliceIterator()
 	txn := s.db.ReadTxn()
 	for id, namespace := range ids {
-		raw, err := txn.First("csi_volumes", "id", namespace, id)
-		if err != nil {
-			return nil, fmt.Errorf("volume lookup failed: %s %v", id, err)
+		if strings.HasPrefix(id, prefix) {
+			raw, err := txn.First("csi_volumes", "id", namespace, id)
```

Comment:
this isn't a change to the previous behavior, but i wonder whether this should also be a `FirstWatch`, so that a blocking query against this endpoint will return if one of the volumes changes. 
this may have been intended by the call here: 
https://github.com/hashicorp/nomad/pull/10158/files#diff-53c7d524897a192ece819816a9dcf006dee46747fbb853e7f638601ea556a72cR2224
but unfortunately `WatchCh()` returns `nil` for `SliceIterator` objects 😢

Reply:
Agreed. Fixed in 5836f3b